### ISSUE TITLE
package_deseq2_1_8_1 fails to install correctly (Ubuntu 14.04)

### ISSUE THREAD

**Issue by bcclaywell** [galaxyproject/tools-iuc#325]
While installation of `package_deseq2_1_8_1` *appears* to install correctly, a problem with `package_r_3_2_1`'s `libgfortran.so` actually prevents installation of at least two of the required R packages, `RcppArmadillo` and `KernSmooth`, which then causes installation of `DESeq2` itself to fail. Excerpts from the installation log:

```
** testing if installed package can be loaded
install_environment.STDOUT DEBUG 2015-10-13 20:54:14,000 Error: loading failed
Error in dyn.load(file, DLLpath = DLLpath, ...) :
  unable to load shared object '/mnt/silver/galaxy/containers/galaxy-qa/export/tool_deps/deseq2/1.8.1/iuc/package_deseq2_1_8_1/245e19b70f1d/RcppArmadillo/libs/RcppArmadillo.so':
  /mnt/silver/galaxy/containers/galaxy-qa/tool_deps/R/3.2.1/iuc/package_r_3_2_1/e46a7803f17b/lib/R/lib/../lib/libgfortran.so.3: version `GFORTRAN_1.4' not found (required by /usr/lib/libblas.so.3)
Execution halted
Error: loading failed
ERROR: loading failed
* removing '/mnt/silver/galaxy/containers/galaxy-qa/export/tool_deps/deseq2/1.8.1/iuc/package_deseq2_1_8_1/245e19b70f1d/RcppArmadillo'
Warning message:
In install.packages(c("RcppArmadillo_0.5.500.2.0.tar.gz"), lib = "/mnt/silver/galaxy/containers/galaxy-qa/tool_deps/deseq2/1.8.1/iuc/package_deseq2_1_8_1/245e19b70f1d",  :
  installation of package 'RcppArmadillo_0.5.500.2.0.tar.gz' had non-zero exit status
```

```
** testing if installed package can be loaded
install_environment.STDOUT DEBUG 2015-10-13 02:42:43,201 Execution halted
Error in dyn.load(file, DLLpath = DLLpath, ...) :
  unable to load shared object '/mnt/silver/galaxy/containers/galaxy-qa/export/tool_deps/deseq2/1.8.1/iuc/package_deseq2_1_8_1/245e19b70f1d/KernSmooth/libs/KernSmooth.so':
  /mnt/silver/galaxy/containers/galaxy-qa/tool_deps/R/3.2.1/iuc/package_r_3_2_1/e46a7803f17b/lib/R/lib/../lib/libgfortran.so.3: version `GFORTRAN_1.4' not found (required by /usr/lib/libblas.so.3)
Error: loading failed
Execution halted
ERROR: loading failed
* removing '/mnt/silver/galaxy/containers/galaxy-qa/export/tool_deps/deseq2/1.8.1/iuc/package_deseq2_1_8_1/245e19b70f1d/KernSmooth'
Warning message:
In install.packages(c("KernSmooth_2.23-13.tar.gz"), lib = "/mnt/silver/galaxy/containers/galaxy-qa/tool_deps/deseq2/1.8.1/iuc/package_deseq2_1_8_1/245e19b70f1d",  :
  installation of package 'KernSmooth_2.23-13.tar.gz' had non-zero exit status
```

```
ERROR: dependency 'RcppArmadillo' is not available for package 'DESeq2'
* removing '/mnt/silver/galaxy/containers/galaxy-qa/export/tool_deps/deseq2/1.8.1/iuc/package_deseq2_1_8_1/245e19b70f1d/DESeq2'
Warning message:
In install.packages(c("DESeq2_1.8.1.tar.gz"), lib = "/mnt/silver/galaxy/containers/galaxy-qa/tool_deps/deseq2/1.8.1/iuc/package_deseq2_1_8_1/245e19b70f1d",  :
  installation of package 'DESeq2_1.8.1.tar.gz' had non-zero exit status
```

**Comment by yhoogstrate** [comment]
@bcclaywell Thaks for reporting :). This makes me realize DESeq 1.8.2 is not in the main toolshed (https://github.com/galaxyproject/tools-iuc/pull/294). I am updating this right now.
@bgruening Can you help on this one?

**Comment by mvdbeek** [comment]
@yhoogstrate @bgruening This is probably related to the way the dependencies are specified in https://github.com/galaxyproject/tools-iuc/blob/master/packages/package_r_3_2_1/tool_dependencies.xml
I'm surprised it works on the testtoolshed. Or is this a recent change on the toolshed codebase?

**Comment by mvdbeek** [comment]
@bcclaywell is there any chance that you could install gfortran on your system and try to install the package again? I think a decision has been made for users having to have gfortran installed. If that doesn't work or gfortran is already installed, what OS are you using?

**Comment by bgruening** [comment]
`libfortran` is shipped with the R binary. So this should work, otherwise it's a bug. atlas and fortran are a pain for me since years :(
@bcclaywell can we tell us the operating system? Are you using the binaries?

**Comment by mvdbeek** [comment]
@bgruening I remember this now: https://github.com/galaxyproject/tools-iuc/issues/190#issuecomment-111555093 
Same trick should work for R, right? Copying the systems libgfortran. If it works for atlas, it should work here,

**Comment by bgruening** [comment]
@mvdbeek exactly, but I don't know if this is the root cause of this problem :(
Maybe we should take this over for all R packages just to be sure?

**Comment by mvdbeek** [comment]
Sounds good. I can do this.

**Comment by bgruening** [comment]
Awesome, thanks!

**Comment by bcclaywell** [comment]
@mvdbeek gfortran was actually already installed on the system, and replacing the libgfortran shared objects in the `package_r_3_2_1` lib directory with symlinks to the system libraries solved the issue with the DESeq2 build.

@bgruening This is on Ubuntu 14.04.

**Comment by bgruening** [comment]
@bcclaywell I bet you can also remove the symlink. As @mvdbeek pointed out we have a work around for newer versions of libfortran. So we can fix this hopefully soon.

**Comment by mvdbeek** [comment]
Closing this as @bcclaywell confirmed that the issue was solved with PR #335
